### PR TITLE
SLING-9688 - Upgrade Sling superimposing to use OSGi R7 annotations

### PR DESCRIPTION
**Pull request by saurabh-khare**
Please refer ticket SLING-9688 for more information
https://issues.apache.org/jira/browse/SLING-9688

### CODE REVIEW

File: bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Bundle-Category: sling
```
Comment (kwin) [inline bnd.bnd:1]: Please don't duplicate meta information already set in parent

File: pom.xml [hunk 2]
```diff
@@ -19,14 +19,13 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.apache.sling</groupId>
-        <artifactId>sling</artifactId>
-        <version>26</version>
-        <relativePath/>
+        <artifactId>sling-bundle-parent</artifactId>
+        <version>38</version>
```
Comment (kwin) [inline pom.xml:23]: Please use most recent version (39)

File: pom.xml [hunk 3]
```diff
@@ -43,120 +42,120 @@
...
-
         </plugins>
     </build>
 
     <dependencies>
         <dependency>
             <groupId>org.apache.sling</groupId>
             <artifactId>org.apache.sling.api</artifactId>
-            <version>2.2.0</version>
+            <version>2.18.4</version>
```
Comment (kwin) [inline pom.xml:66]: Why do you require new runtime dependencies (api, commons osgi, ...)?

File: pom.xml [hunk 4]
```diff
@@ -43,120 +42,120 @@
...
-        </dependency>
-        <dependency>
-            <groupId>commons-collections</groupId>
-            <artifactId>commons-collections</artifactId>
-            <version>3.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.service.metatype.annotations</artifactId>
+            <version>1.4.0</version>
```
Comment (kwin) [inline pom.xml:106]: Don't set versions for dependencies managed in parent

**Review by kwin — CHANGES_REQUESTED** [review]

File: bnd.bnd (hunk 1 above)
Comment (saurabh-khare) [reply bnd.bnd]: Removed duplicated information

File: pom.xml (hunk 2 above)
Comment (saurabh-khare) [reply pom.xml]: Updated to version 39

File: pom.xml (hunk 3 above)
Comment (saurabh-khare) [reply pom.xml]: Reverted to previous versions for runtime dependencies

File: pom.xml (hunk 4 above)
Comment (saurabh-khare) [reply pom.xml]: Removed

**Comment by saurabh-khare** [comment]
@kwin Can you please review again ?

**Comment by rombert** [comment]
@kwin - did you have time to re-review this?

**Comment by kwin** [comment]
@rombert Sorry, this slipped through. I am gonna pick it up from where it is and apply some more updates (nowadays we are at Sling Parent 46).

**Comment by kwin** [comment]
Superseded by #3

**Pull request closed without merging**